lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1574,12 +1589,19 @@ void IRGenDebugInfoImpl::setCurrentLoc(IRBuilder &Builder,
     // Reuse the last source location if we are still in the same
     // scope to get a more contiguous line table.
     L = LastDebugLoc;
+  } else if (DS == LastScope && Loc.is<ArtificialUnreachableLocation>() &&
+             Opts.DebugInfoFormat == IRGenDebugInfoFormat::CodeView) {
+    // Remove artificial unreachable locations to make CodeView linetables
+    // more contiguous.
+    L = LastDebugLoc;
   } else {
     // Decode the location.
     L = getDebugLocation(Loc);
     // Otherwise use a line 0 artificial location, but the file from the
-    // location.
-    if (Loc.isAutoGenerated()) {
+    // location. If we are emitting CodeView, we want to avoid using line 0
+    // for cleaner linetables.
```

Comment:
Please don't use fuzzy wording such as "cleaner" and instead spell out what the problem is and why attributing it to a different source line is preferable to having an unknown location.